mobify/platform-scaffold
# [WEB-1380] Implement Tree Shaking

**JIRA**: [[WEB-1380] Implement Tree Shaking](https://mobify.atlassian.net/browse/WEB-1380)
 **Linked PRs**: [astro#807](https://github.com/mobify/astro/pull/807)

## Changes
- Modify `update-web-astro-client.sh` script to pull es6 modules from `progressive-app-sdk`
- Set a babel configuration to stop the transpiler from transforming modules to a different type.

## Todos
- [x] Get Astro PR merged
- [x] Point to a published version of Astro... when that happens (or #develop branch when the linked PR gets merged)

## How to test-drive this PR
1. Run `rm -rf node_modules && npm i` inside the `native` directory.
1. Execute `./scripts/update-web-astro-client.sh`
    1. `web/app/vendor/astro-detect.js` should now be an es6 module
1. Now in `/web` run `npm run dev` (chunks named [template-name].js should now be slightly smaller than before)
1.  Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
1. Site should still load without complaints.

## Results
![dev-tree-shaking-impacts](https://user-images.githubusercontent.com/10932540/27206355-e03d3e46-51eb-11e7-98cc-8d16f2f0c4c3.png)

1.js : 46.2kb → 36.6
2.js : 64.4kb → 52.3
3.js : 62.6kb → 50.9
4.js : 34.4kb → 27.5
5.js : 38.1kb → 31.5
6.js : 40.6kb → 32.7
7.js : 29.9kb → 23.5
main.js : 412kb → 324kb 
**Total 149.2kb reduction → 20.4%**
